sceptre/diffing/stack_differ.py

```diff
@@ -0,0 +1,234 @@
+                    # dependent upon doesn't exist, this one most likely won't exist either, so
+                    # when deployed version is compared to this one, it will indicate we're creating
+                    # a new stack where one doesn't exist. The only place this value would be shown
+                    # is where the current stack DOES exist, but a new dependency is added that does
+                    # not exist yet. In which case, it will show the most usable output we can
+                    # provide right now.
+                    value_to_use = f'!{type(value).__name__}'
+                    if value.argument is not None:
+                        value_to_use += f': {value.argument}'
+            else:
+                value_to_use = value
+            parameters[key] = value_to_use
+
+        return parameters
+
+    def _create_deployed_stack_config(self, stack_actions: StackActions) -> Optional[StackConfiguration]:
+        description = stack_actions.describe()
+        if description is None:
+            return None
+
+        stacks = description['Stacks']
+        for stack in stacks:
+            return StackConfiguration(
+                parameters={
+                    param['ParameterKey']: param.get('ResolvedValue', param['ParameterValue'])
+                    for param
+                    in stack.get('Parameters', [])
+                },
+                tags={
+                    tag['Key']: tag['Value']
+                    for tag in stack['Tags']
+                },
+                name=stack['StackName'],
+                notification_arns=stack['NotificationARNs'],
+                role_arn=stack.get('RoleARN')
+            )
+
+    def _get_deployed_template(self, stack_actions: StackActions) -> str:
+        template_string = stack_actions.fetch_remote_template()
+
+        if template_string is None:
+            return '{}'
+
+        return template_string
+
+    @abstractmethod
+    def compare_templates(
+        self,
+        deployed: str,
+        generated: str,
+    ) -> DiffType:
+        """Implement this method to return the diff for the templates"""
+
+    @abstractmethod
+    def compare_stack_configurations(
+        self,
+        deployed: Optional[StackConfiguration],
+        generated: StackConfiguration,
+    ) -> DiffType:
+        """Implement this method to return the diff for the Stack Configurations"""
+
+
+class DeepDiffStackDiffer(StackDiffer):
+    def __init__(
+        self,
+        *,
+        universal_template_loader: Callable[[str], Tuple[dict, str]] = cfn_flip.load
+    ):
+        self.load_template = universal_template_loader
+
+    def compare_stack_configurations(
+        self,
+        deployed: Optional[StackConfiguration],
+        generated: StackConfiguration,
+    ) -> deepdiff.DeepDiff:
+        return deepdiff.DeepDiff(deployed, generated, verbose_level=2)
```

Comment:
hmm, would be nice to allow user to set `verbose_level` although i'm not sure if that makes sense.  Maybe leave as an improvement if someone really cares.

Reply:
With DeepDiff, verbose level 1 shows what _keys_ have been changed, but it doesn't show what that change actually was. Verbose level 2 will show the actual before and after values. Given that, I don't really think that any other verbose level would serve the purpose of the diff command.